packages/apollo-link-dedup/src/__tests__/dedupLink.ts

```diff
@@ -268,3 +268,45 @@ describe('DedupLink', () => {
     expect(unsubscribed).toBe(true);
   });
 });
+
+it(`should be wait all observers unsubscribe`, () => {
```

Comment:
I added a very similar test in [this commit](https://github.com/apollographql/apollo-client/pull/4576/commits/3c40d984a78a1bc0fc6c7957d43b1132d9f8f85a#diff-0a0243849ea07e7f9b27a68afb1cd653), in case you want to check it out.

Reply:
This is great. :stuck_out_tongue_winking_eye: